bliss/catalog.py

```diff
@@ -489,7 +489,13 @@ def to_tile_params(
 
         for ii in range(self.batch_size):
             n_sources = int(self.n_sources[ii].item())
-            source_indices = tile_coords[ii][:n_sources]
+            plocs_ii = self.plocs[ii][:n_sources]
+            x0_mask = (plocs_ii[:, 0] > 0) & (plocs_ii[:, 0] < self.height)
+            x1_mask = (plocs_ii[:, 1] > 0) & (plocs_ii[:, 1] < self.width)
+            x_mask = x0_mask * x1_mask
+            n_filter_sources = x_mask.sum()
```

Comment:
I think it'd be better to do the filtering only on demand: better to give users and error if they try to convert catalog formats without first removing out-of-bounds locations. Would you a new method called something like `filter_out_of_bounds`? And then call that method following data augmentation?

Reply:
Here's an alternative that may be less tricky: add a keyword argument to `to_tile_coords`  called `filter_oob` which is false by default. Only if it's true will we filter light sources that are out of bounds (either too negative or too large).